### DISCUSSION TITLE
Immich ML, Thumbnail generation, etc on ARM

### DISCUSSION THREAD

**Discussion by mufeedali** [immich-app/immich#4857] [Q&A] [answered]
Hi,

I have an ARM VPS where I want to use Immich. It's a pretty powerful Ampere-based server. ML and every other feature seemed to be working fine until I mass uploaded 2700 files. Since then, all the jobs seem to be eternally running, with the waiting counts constantly increasing, and in the case of thumbnail generation, way beyond the number of items actually in Immich.

![image](https://github.com/immich-app/immich/assets/28823274/4ca95c04-846c-4f1b-bbfd-3abd999db65c)

Looking through GitHub, I can't see anyone encountering issues like this. So could this be due to ARM? I saw some issues where it's indicated that no AVX means no machine learning. Is that still true?

Thanks :)

EDIT: To add, the upload was via the Web UI and there were errors and there seems to be partial uploads. Maybe that could be the issue in stead?

**Comment by bo0tzz** [comment, marked as answer]
ARM should work fine, the demo instance and some developer environments run on an ARM server just fine. 
There are multiple thumbnails generated per asset, so for the count to be larger than you might expect is normal.

**Reply by mufeedali** [reply]
Thanks a lot! You were right, it's all good now :)
